build.gradle.kts

```diff
@@ -0,0 +1,663 @@
+            checkers.addAll(listOf(
+                    "org.checkerframework.checker.index.IndexChecker",
+                    "org.checkerframework.checker.interning.InterningChecker",
+                    "org.checkerframework.checker.optional.OptionalChecker",
+            ))
+        }
+        extraJavacArgs = mutableListOf(
+                "-AassumeAssertionsAreEnabled",
+                "-AinvariantArrays",
+                "-Alint=cast:redundant,cast:unsafe",
+                "-ArequirePrefixInWarningSuppressions",
+                "-AshowSuppressWarningsStrings",
+                "-Astubs=$rootDir/config/checker/",
+                // The `-AstubWarnIfNotFoundIgnoresClasses` flag is required since Checker 3.14.0,
+                // the version since which `-AstubWarnIfNotFound` is assumed to be true for custom stub files.
+                // Without this flag, we would end up with a lot of errors in subprojects where any of the stubbed libraries is NOT on the classpath:
+                // e.g. compilation of a subproject where Vavr is NOT a dependency would fail
+                // since for simplicity, we're providing the same set of stubs to Checker in each subproject
+                // (`$rootDir/config/checker/`, which includes e.g. Vavr).
+                "-AstubWarnIfNotFoundIgnoresClasses",
+                "-AsuppressWarnings=allcheckers:type.anno.before.decl.anno,allcheckers:type.anno.before.modifier,allcheckers:type.checking.not.run,value:annotation",
+        )
+        dependencies {
+            compileOnly(rootProject.libs.checker.qual)
+            add("checkerFramework", rootProject.libs.checker)
+        }
+    }
+
+
+    apply<CheckstylePlugin>()
+    configure<CheckstyleExtension> {
+        tasks.named<Checkstyle>("checkstyleTest") {
+            enabled = false
+        }
+
+        configProperties = mapOf("rootCheckstyleConfigDir" to "$rootDir/config/checkstyle")
+    }
+
+
+    // A few libraries (like JGit and reflections) transitively pull in a version of slf4j-api
+    // that might be different from the slf4j-api version that IntelliJ depends on.
+    // SLF4J guarantees that the code compiled against a certain slf4j-api version will work with any other version of slf4j-api
+    // (http://www.slf4j.org/manual.html#compatibility).
+    // We rely on that guarantee: our plugin effectively uses whatever slf4j-api version is provided by IntelliJ.
+    // SLF4J does NOT guarantee, however, that slf4j-api version X will work with any slf4j implementation version Y for X != Y.
+    // To avoid a clash between JGit&co.'s slf4j-api and Intellij's slf4j implementation
+    // (and also between JGit&co.'s slf4j-api and Intellij's slf4j-api), we need to exclude the former from ALL dependencies.
+    configurations.runtimeClasspath {
+        exclude(group = "org.slf4j", module = "slf4j-api")
+    }
+
+    val addIntellijToCompileClasspath by extra {
```

Comment:
Huh okay... so this is the sole of the `extra`s that remains used as an `extra` (and not temporarily inlined)... is there a specific reason for that?